lib/cloud/aws/identity.go

```diff
@@ -106,6 +108,10 @@ func (i identityBase) String() string {
 	return i.arn.String()
 }
 
+type callerIdentityGetter interface {
+	GetCallerIdentity(ctx context.Context, params *stsV2.GetCallerIdentityInput, optFns ...func(*stsV2.Options)) (*stsV2.GetCallerIdentityOutput, error)
+}
+
 // GetIdentityWithClient determines AWS identity of this Teleport process
 // using the provided STS API client.
 func GetIdentityWithClient(ctx context.Context, stsClient stsiface.STSAPI) (Identity, error) {
```

Comment:
How long would it take to get rid of STS v1 in its entirety?

Reply:
I'm not sure I only touched this package because it was being used by lib/configurators/aws

Probably would not take very long, but I'd rather not include it in this PR